### PR TITLE
ci: Simplify the build process

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

In this change we introduce `@guardian/node-riffraff-artifact` which allows us to remove a load of bash that massages files into a directory structure that Riff-Raff understands. That is, we remove some code 🎉.

The build has also been simplified, dropping the `aws-sdk` dependency from the artifact as it is included in the AWS Lambda runtime already. This also results in a smaller artifact.

### Before
image

### After
image

## What testing has been performed for this change?

I've deployed this branch to the DevX account and can continue to see logs.

image

## How can we measure success?

It's easier to update CI (which we will be with the addition of CDK to the project).

## Have we considered potential risks?

n/a

### CODE REVIEW

File: riff-raff.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ templates:
     contentDirectory: cloudwatch-logs-management
     parameters:
       bucketSsmLookup: true
-      fileName: lambda.zip
+      fileName: cloudwatch-logs-management.zip
```
Comment (akash1810) [inline riff-raff.yaml:26]: `@guardian/node-riffraff-artifact` (sensibly IMO!) will always name a file `action/action.zip` as described in the  `artifact.json` file.

Here we bring `riff-raff.yaml` inline with this naming.

File: template.yaml [hunk 2]
```diff
@@ -91,7 +91,7 @@ Resources:
     Properties:
       CodeUri:
         Bucket: !Ref DistBucket
-        Key: !Sub ${Stack}/${Stage}/set-retention/lambda.zip
+        Key: !Sub ${Stack}/${Stage}/set-retention/cloudwatch-logs-management.zip
```
Comment (akash1810) [inline template.yaml:94]: `@guardian/node-riffraff-artifact` (sensibly IMO!) will always name a file `action/action.zip` as described in the  `artifact.json` file.

Here (and below) we bring the CFN inline with this naming.

File: scripts/ci.sh [hunk 3]
```diff
@@ -1,55 +1,30 @@
 #!/bin/bash
```
Comment (akash1810) [inline scripts/ci.sh:1]: Quite a lot of changes in this file, but that goes some way to illustrating the benefit of this PR 😅 .

**Review by jacobwinch — APPROVED** [review]
Nice work 👍

**Pull request merged**